packages/react/src/auto/AutoForm.ts

```diff
@@ -58,7 +58,14 @@ export const useFormFields = (
   return useMemo(() => {
     if (!metadata) return [];
     const action = isActionMetadata(metadata) ? metadata.action : metadata;
-    const objectFields = action.inputFields.filter((field) => field.configuration.__typename === "GadgetObjectFieldConfig");
+    const objectFields =
+      metadata.__typename === "GadgetGlobalAction"
+        ? []
+        : action.inputFields.filter(
+            (field) =>
+              field.configuration.__typename === "GadgetObjectFieldConfig" &&
+              (metadata.__typename === "GadgetModel" ? field.apiIdentifier === metadata.apiIdentifier : true)
```

Comment:
NIT: `metadata.__typename === "GadgetModel"` could be a const outside of the filter. Also seems to have similar intention to the above `metadata.__typename === "GadgetGlobalAction"` and `isActionMetadata(metadata)`